tests/periph_pwm/main.c

```diff
@@ -68,12 +167,94 @@ int main(void)
         }
 
         state += step;
-        if (state <= 0 || state >= (int)STEPS) {
+        if (state <= 0 || state >= (int)OSC_STEPS) {
             step = -step;
         }
 
-        xtimer_periodic_wakeup(&last_wakeup, INTERVAL);
+        xtimer_periodic_wakeup(&last_wakeup, OSC_INTERVAL);
     }
 
     return 0;
 }
+
+int power(int argc, char** argv){
+    if (argc != 3) {
+        printf("usage: %s <dev> <state>\n", argv[0]);
+        printf("\tdev: device by number between 0 and %d\n", PWM_NUMOF-1);
+        printf("\tstate:\n");
+        printf("\t\t0: power off\n");
+        printf("\t\t1: power on\n");
+        return 1;
+    }
+
+    errno = 0;
+    uint32_t dev = strtol(argv[1], NULL, 10);
+    if (errno != 0 || dev >= PWM_NUMOF) {
+        printf("[pwm] Error: device %s is unknown.\n", argv[1]);
+        return 1;
+    }
+
+    errno = 0;
+    uint32_t state = strtol(argv[2], NULL, 10);
+    if (errno == 0) {
+        switch (state) {
+            case (0):
+                printf("Powering down PWM device.\n");
+                pwm_poweroff(dev);
+                return 0;
+            case (1):
+                printf("Powering up PWM device.\n");
+                pwm_poweron(dev);
+                return 0;
+            default:
+                break;
+        }
+    }
+    printf("[pwm] Error: power state %s not available.", argv[2]);
+    return 1;
+
+}
+
+int power_test(int argc, char** argv){
+    if (argc != 2) {
+        printf("usage: %s <dev>\n", argv[0]);
+        printf("\tdev: device by number between 0 and %d\n", PWM_NUMOF-1);
+        return 1;
+    }
+
+    uint32_t dev = strtol(argv[1], NULL, 10);
+    if (dev >= PWM_NUMOF) {
+        printf("[pwm] Error: device %ld is unknown.\n", dev);
+        return 1;
+    }
+
+    printf("Powering down PWM device and sleeping for %d second(s)...\n",
+           PWR_SLEEP);
+    pwm_poweroff(dev);
```

Comment:
you should not use the numeric PWM identifier directly, but always use the 'access marcro' `PWM_DEV(dev)`